app/controllers/project.py

```diff
@@ -863,3 +872,62 @@ def post(self, project_id):
             status='success',
             message=f'project {project_id} enabled successfully'
         ), 201
+
+
+class ProjectPinView(Resource):
+    @jwt_required
+    def post(self , project_id):
+
+        current_user_id = get_jwt_identity()
+
+        project_user = ProjectUser.query.filter_by(user_id = current_user_id , project_id = project_id).first()
+
+        if (not project_user):
```

Comment:
Like I highlighted before, please use existing project access logic, otherwise we will be forced to come back and modify this very line if we were to refactor the logic in the entire application
```suggestion
      if not is_authorised_project_user(project, current_user_id, 'member'):
                return dict(status='fail', message='unauthorised'), 403
```